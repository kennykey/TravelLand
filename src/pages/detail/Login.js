"use client"
import { Container,Form,Button } from "react-bootstrap"
import useAuth from "@/useApi/useAuth";
import React ,{ useState } from "react"
import LayOut from "@/component/LayOut";
import { useDispatch, useSelector } from "react-redux";
import Image from "next/image";

export default function Login(){
    const {Auth,loading} = useAuth();
    const {login} = useSelector((state)=>state.auth);
    const dispatch = useDispatch();

    const handleLogin = async (e) =>{
        e.preventDefault();
        const email = e.target.email.value;
        const password = e.target.password.value;

        Auth(
            'login',
            {email,password}
        );
             
        if(!email && !password) return
        dispatch(Login(email,password));
    };

    if(loading){
        return(
            <h2 className="text-center fw-bold">
                Loading...
            </h2>
        )
    }
    console.log(login);
    return ( 
        <LayOut>
            <Container className="d-flex m-5 tengah">
                <Container>
                    <Image src="/sign_in.png" alt="Sign_in" className="img-fluid" height={400} width={400}/>
                </Container>
                <Container className="m-5">
                    <Form onSubmit={handleLogin} style={{width:"400px"}}>
                        <h4 className="text-center fw-bold mb-3">Login</h4>
                        <Form.Group className="mb-3"  controlId="formBasicEmail">
                            <Form.Control type="email" placeholder="Enter Email" name="email" id="email"/>
                        </Form.Group>
                        <Form.Group className="mb-3"  controlId="formBasicPassword">
                            <Form.Control type="password" placeholder="Enter Password" name="password" id="password"/>
                        </Form.Group>
                        <Button variant="primary" type="submit" className="w-100">
                            Submit
                        </Button>
                    </Form>
                </Container>
            </Container>
        </LayOut>
     );
}
 
