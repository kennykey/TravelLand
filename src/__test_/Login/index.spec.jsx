import React from 'react';
import { render } from '@testing-library/react';
import Login from '@/pages/detail/Login';

describe('login page', () => {
  it('should handle render login', () => {
    const { container } = render(<Login />);
    expect(container).toBeInTheDocument();
  });

  it('should handle render input text', () => {
    const { container } = render(<input type="text" label="email"/>);
    expect(container).toMatchSnapshot();
  });

  it('should handle render input text', () => {
    const { container } = render(<input type="text" label="password"/>);
    expect(container).toMatchSnapshot();
  });

  it('should handle render button not disable', () => {
    const { container } = render(
      <button text={'button'} type="submit"/>);
    expect(container).toMatchSnapshot();
  });
});
