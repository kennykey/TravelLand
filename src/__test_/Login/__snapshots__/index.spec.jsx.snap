// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login page should handle render button not disable 1`] = `
<div>
  <button
    text="button"
    type="submit"
  />
</div>
`;

exports[`login page should handle render input text 1`] = `
<div>
  <input
    label="email"
    type="text"
  />
</div>
`;

exports[`login page should handle render input text 2`] = `
<div>
  <input
    label="password"
    type="text"
  />
</div>
`;
